support-frontend/assets/pages/[countryGroupId]/checkout.tsx

```diff
@@ -365,15 +366,27 @@ export function Checkout({ geoId, appConfig }: Props) {
 				? 'Monthly'
 				: 'Annual';
 
+		let fulfilmentOption: FulfilmentOptions;
```

Comment:
I would break the setter out into a separate function so that this can be a const. eg.
```typescript
const fulfilmentOptions = getFulfilmentOptions(productKey);
```

Reply:
Completely agree that mutable state can cause issues.

There is a little tension here in trying not to create early / unnecessary abstractions that might lead to coupling i.e. a `function` that could be used elsewhere. Defining it and not exporting it would work too (which is what I think the suggestion is here).

It's a pattern that's used across JS codebases - as the `switch` / `case` syntax is a little lacking in the sense that you can't run it and assign it to a value as and when. The convention is to not mutate the value outside of the `switch` / `case`.

You can actually run anonymous `switch` statements, but they do look ugly
```ts
const productKey = '...';
const fulfilmentOptions = (() => {
  switch (productKey) {
    case 'SupporterPlus':
    case 'Contribution':
      return 'NoFulfilmentOptions';
    case 'TierThree':
      return countryGroupId === 'International' ? 'RestOfWorld' : 'Domestic';
    default:
      // ToDo: define for every product here
      return 'NoFulfilmentOptions';
  }
})()
```

[Interesting read here](https://github.com/tc39/proposal-pattern-matching) on a better match proposal.

Reply:
Super interesting discussion, I'd suggest let's merge without it and come back to this - because it's a pattern throughout this checkout

Reply:
Have applied Rupert's immutable suggestion/change (private function) so we can get the merge in. As suggested, perhaps we can examine this pattern more closely afterwards as we are running out of time before tomorrow.